File: leetcode/editor/cn/LCP 17_速算机器人_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func Test_givenNormal_whenLCP17_thenSuccess(t *testing.T) {
	assert.Equal(t, 4, calculate("AB"), "shouldEqual")
}

//leetcode submit region begin(Prohibit modification and deletion)
func calculate(s string) int {
	x, y := 1, 0
	for _, c := range s {
		switch c {
		case 'A':
			x = x*2 + y
		case 'B':
			y = y*2 + x
		default:
		}
	}
	return x + y
}

//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/cn/LCP 17_速算机器人_test.py
import unittest


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        self.assertEqual(self.solution.calculate('AB'), 4)


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def calculate(self, s: str) -> int:
        x, y = 1, 0
        for op in s:
            if op == 'A':
                x = x * 2 + y
            else:
                # op == 'B'
                y = y * 2 + x
        return x + y


# leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()
